src/modules/wlr/taskbar.cpp

```diff
@@ -383,6 +390,15 @@ void Task::handle_app_id(const char *app_id)
 {
     app_id_ = app_id;
 
+    Glib::RefPtr<Gio::DesktopAppInfo> app_info = gen_app_info(app_id);
```

Comment:
In case missing icon and name options it will cost unnecessary overhead. And it will be searched twice, as app_info was't saved.